### DISCUSSION TITLE
Can cost tracker work with a total_increasing energy sensor that resets daily?

### DISCUSSION THREAD

**Discussion by billmichell** [BottlecapDave/HomeAssistant-OctopusEnergy#747] [Q&A]
My best grid import sensor is a daily accumulating energy sensor on my battery inverter, of type total_increasing.

I configured a Cost Tracker for it with entity_accumulative_value set to true, as that seemed to be right from the documentation, and gave me good numbers for the day, I even figured out how to disable it while exporting.

But then midnight came, and it saw the reset to 0 as a massive momentary export event.

Is there some combination of settimgs I can do, or some template (or other) entity I can create that will get a nice reading?

**Comment by asuweb** [comment]
I'm seeing the exact same issue as you are.

**Comment by billmichell** [comment]
OK, so I more-or-less fixed this by creating a new Utility meter entity and adding my daily increasing sensor as its primary input. The Utility meter knows that a reset to 0 doesn’t need tracking, and produces an absolutely increasing output. This can then be fed into the cost tracker. Or rather, a brand new cost tracker as I couldn’t find a way to switch the input for the original one.

**Reply by asuweb** [reply]
Yes I was thinking along the same lines.  I assume you set the "Meter Reset Cycle" to "No Cycle" for the utility meter?

**Reply by billmichell** [reply]
Correct

**Comment by BottlecapDave** [comment]
I'm sorry that you're having issues with the cost tracker. If there is an issue with certain types of sensors, please raise an issue. Ideally this type of sensor should work out of the box without much faffing. Screenshots of the behavior and the attributes of the offending sensor from the developer tools will help in the assistance of recreation and testing of the fix.

**Reply by billmichell** [reply]
Unfortunately I deleted the dodgy tracker, so I can’t show you. The MQTT configuration for the sensor looks like this:
```yaml
    - name: "Electricity Consumption Today"
      state_topic: "sofar/today_consumption"
      expire_after: 120
      state_class: "total_increasing"
      device_class: "energy"
      unit_of_measurement: "kWh"
```

`total_increasing` means it rises slowly but occasionally resets to zero. The expected behaviour is that this isn’t seen as a large occasional negative delta. Unfortunately, the cost tracker just puts a big negative balance onto the running total. Feeding it through a Utility Meter first fixes the problem.

**Reply by asuweb** [reply]
I've set up the cost trackers again and will report back in a day or so

**Reply by asuweb** [reply]
Issue #759
